### PR TITLE
Added DATETIME_DIFF in SqlLibraryOperators

### PR DESCRIPTION
**Pull request by vinayujeedm**
https://datametica.atlassian.net/browse/RNBS-593

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -6197,6 +6198,24 @@ private void checkLiteral2(String expression, String expected) {
     assertThat(toSql(root, DatabaseProduct.HIVE.getDialect()), isLinux(expectedHive));
   }
 
+  @Test public void testDateTimeDiffFunctionRelToSql() {
+    final RelBuilder builder = relBuilder();
+    final RexNode DateTimeDiffRexNode = builder.call(SqlLibraryOperators.DATETIME_DIFF,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6203]: change the case to camelCase

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -6197,6 +6198,24 @@ private void checkLiteral2(String expression, String expected) {
...
+        builder.call(SqlStdOperatorTable.CURRENT_DATE),
+        builder.call(SqlStdOperatorTable.CURRENT_DATE), builder.literal(TimeUnit.HOUR));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(DateTimeDiffRexNode, "DT"))
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6208]: you can keep alias name more meaningful like hours in this case.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (vinayujeedm) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Done

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 2 above)
Comment (vinayujeedm) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Done

**Review by nosheenSheikh02 — APPROVED** [review]

**Review by anand-dm — APPROVED** [review]

**Pull request merged**